File: go/242_Valid_Anagram/242_Valid_Anagram.go
func isAnagram(s string, t string) bool {
    ss := make([]int, 26)
    if len(s) != len(t) {
        return false
    }
    for i, l := range s {
        ss[l-'a'] += 1
        ss[t[i]-'a'] -= 1
    }
    for _, l := range ss {
        if l != 0 {
            return false
        }
    }
    return true
}

File: java/242_Valid_Anagram/242_Valid_Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] ss = new int[26];
        if (s.length() != t.length()) {
            return false;
        }
        for (int i = 0; i < s.length(); i++) {
            ss[s.charAt(i)-'a'] += 1;
            ss[t.charAt(i)-'a'] -= 1;
        }
        for (int i : ss) {
            if (i != 0) {
                return false;
            }
        }
        return true;
    }
}
